### PR TITLE
update gradle to 8.5

### PR DESCRIPTION
**Pull request by kishansairam9**

### CODE REVIEW

**Comment by kotharironak** [comment]
We don't need it here right now.

File: build.gradle.kts [hunk 1]
```diff
@@ -21,9 +21,6 @@ subprojects {
 
   pluginManager.withPlugin("java") {
     configure<JavaPluginExtension> {
-      sourceCompatibility = JavaVersion.VERSION_11
```
Comment (aaron-steinfeld) [inline build.gradle.kts:24]: You can leave this as-is.

Comment (kishansairam9) [reply build.gradle.kts:24]: Is it a problem if I remove it though? Do we default to java 11 in CI

Comment (aaron-steinfeld) [reply build.gradle.kts:24]: If removed it just means it defaults to the JVM of the environment it runs in. This should always have been specified in some way (toolchain, in modern gradle) but is missing in a lot of projects. The default in CI is java 11, yes. But we don't want to rely on that assumption for every developer's machine. In other words, this allows a user to run this build with a later jdk without issue.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**